spock-core/src/main/java/org/spockframework/compiler/AbstractDeepBlockRewriter.java

```diff
@@ -22,6 +22,7 @@
 import org.codehaus.groovy.ast.stmt.*;
 
 public class AbstractDeepBlockRewriter extends StatementReplacingVisitorSupport {
+  private final AstNodeCache nodeCache;
   protected Block block;
```

Comment:
Could `block` be also made `final` along the way?

Reply:
I guess not? It is reassigned in first line of `visit(Block block)`

Reply:
Ok, in the GH review view not everything is visible.